Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/migration/CoronaTestMigration.kt

```diff
@@ -0,0 +1,85 @@
+import timber.log.Timber
+import javax.inject.Inject
+
+@Reusable
+class CoronaTestMigration @Inject constructor(
+    private val submissionSettings: SubmissionSettings,
+    private val tracingSettings: TracingSettings,
+) {
+    suspend fun startMigration(): Set<CoronaTest> {
+        val token: RegistrationToken? = submissionSettings.registrationToken.value
+        if (token == null) {
+            Timber.tag(TAG).d("Nothing to migrate, token was null.")
+            return emptySet()
+        } else {
+            Timber.tag(TAG).i("Migrating token %s", token)
+        }
+
+        val devicePairingSuccessfulAt = submissionSettings.devicePairingSuccessfulAt
+        Timber.tag(TAG).v("devicePairingSuccessfulAt=%s", devicePairingSuccessfulAt)
+        if (devicePairingSuccessfulAt == null) {
+            if (CWADebug.isDeviceForTestersBuild) {
+                throw IllegalStateException("Can't have registration token without device pairing timestamp !?")
+            }
+            return emptySet()
+        }
+
+        val isAllowedToSubmitKeys = submissionSettings.isAllowedToSubmitKeys
+        Timber.tag(TAG).v("isAllowedToSubmitKeys=%s", isAllowedToSubmitKeys)
+
+        val isSubmissionSuccessful = submissionSettings.isSubmissionSuccessful
+        Timber.tag(TAG).v("isSubmissionSuccessful=%s", isSubmissionSuccessful)
+
+        val hasViewedTestResult = submissionSettings.hasViewedTestResult.value
+        Timber.tag(TAG).v("hasViewedTestResult=%s", hasViewedTestResult)
+
+        val hasGivenConsent = submissionSettings.hasGivenConsent.value
+        Timber.tag(TAG).v("hasGivenConsent=%s", hasGivenConsent)
+
+        // TODO per test ?
+        val testResultNotificationSent = tracingSettings.isTestResultAvailableNotificationSent
+        Timber.tag(TAG).v("testResultNotificationSent=%s", testResultNotificationSent)
+
+        val legacyPCRTest = PCRCoronaTest(
+            testGUID = LEGACY_GUID,
+            registrationToken = token,
+            registeredAt = devicePairingSuccessfulAt,
+            testResult = when (isAllowedToSubmitKeys) {
+                true -> CoronaTestResult.PCR_POSITIVE
+                else -> CoronaTestResult.PCR_OR_RAT_PENDING
+            },
+            isSubmitted = isSubmissionSuccessful,
+            isViewed = hasViewedTestResult,
+            isAdvancedConsentGiven = hasGivenConsent,
+            isNotificationSent = testResultNotificationSent,
+        )
+        return setOf(legacyPCRTest).also {
+            Timber.tag(TAG).d("Offering converted legacy CoronaTest: %s", it)
+        }
+    }
+
+    suspend fun finishMigration() {
+        Timber.tag(TAG).i("finishMigration()")
+        submissionSettings.deleteLegacyTestData()
+        tracingSettings.deleteLegacyTestData()
+    }
+
+    companion object {
+        private const val TAG = "CoronaTestMigration"
+
+        /**
+         * We only use this for identification
+         */
+        private const val LEGACY_GUID = "legacy-guid"
```

Comment:
"legacy-pcr-guid" ?  As I understand this GUID is used  the HotFlow Map , it is better to distinguish it as PCR

Reply:
The only mandatory requirement for it, is to be different from an actual corona test guid (which are UUIDs). I'll use your suggestion though.